CMakeLists.txt

```diff
@@ -70,6 +70,7 @@ set(PYBIND11_HEADERS
   include/pybind11/stl.h
   include/pybind11/stl_bind.h
   include/pybind11/typeid.h
+  include/pybind11/detail/class_.h
```

Comment:
Many of these header files are actually implementation details, so the separation wrt. ``class_.h`` seems artificial. I'd either leave it at the top level (and rename to something without a trailing underscore like ``class_support.h``) or refactor the header files more significantly (though that doesn't necessarily need to be in this commit).